batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/dao/StudentDAO.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class StudentDAO {
+  private static final Logger LOGGER = Logger.getLogger(StudentDAO.class.getName());
+
+  public void insert(Student s) throws DataException//get the object
+  {
+    try (Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.INSERT);) {
+      pstmt.setString(1, s.getFirstName());
+      pstmt.setString(2, s.getMiddleName());
+      pstmt.setString(3, s.getLastName());
+      pstmt.setString(4, s.getAddress());
+      pstmt.setInt(5, s.getGrade());
+      pstmt.executeUpdate();
+      conn.close();
+    } catch (SQLException ex) {
+      LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+      throw new DataException();
+    }
+  }
+
+  public List<Student> fetch(int page, int limit) throws DataException {
+    try(Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.LIST)) {
+      List<Student> stdList = new ArrayList<Student>();
+      int startOffset = (page - 1) * limit;
+      pstmt.setInt(1, limit);
+      pstmt.setInt(2, startOffset);
+      ResultSet rs = pstmt.executeQuery();
+      while (rs.next()) {
+        Student std = new Student();
+        std.setId(rs.getInt(1));
+        std.setFirstName(rs.getString(2));
+        std.setMiddleName(rs.getString(3));
+        std.setLastName(rs.getString(4));
+        std.setAddress(rs.getString(5));
+        std.setGrade(rs.getInt(6));
+        stdList.add(std);
+      }
+      conn.close();
+      return stdList;
+    } catch (SQLException ex) {
+      LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+      throw new DataException();
+    }
+  }
+
+  public void delete(int id) throws DataException//get the object
+  {
+    try (Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.DELETE)){
+      pstmt.setInt(1, id);
+      pstmt.executeUpdate();
+    } catch (SQLException ex) {
+      LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+      throw new DataException();
+    }
+  }
+
+  public Student fetchById(int id) throws DataException {
+    try (Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.SELECT_BY_ID)){
+      pstmt.setInt(1, id);
+      ResultSet rs = pstmt.executeQuery();
+      Student std = null;
+      while (rs.next()) {
+        std = new Student();
+        std.setId(rs.getInt(1));
+        std.setFirstName(rs.getString(2));
+        std.setMiddleName(rs.getString(3));
+        std.setLastName(rs.getString(4));
+        std.setAddress(rs.getString(5));
+        std.setGrade(rs.getInt(6));
+      }
+      return std;
+    } catch (SQLException ex) {
+      LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+      throw new DataException();
+    }
+  }
+
+  public void edit(Student s) throws DataException {
+    try(Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.UPDATE)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)